### PR TITLE
ROCANA-10063: sigar tests pass on all supported platforms

### PR DESCRIPTION
**Pull request by adamlamar**
This PR makes `go test` pass on all supported platforms - windows, linux, darwin.

Unfortunately it seems the best way to accomplish this is to run the `Get()` methods in `sigar_interface_test.go` and detect when an error has occurred for a known unsupported platform. This means we pepper the tests with checks against `runtime.GOOS`, but I think this is the best way:

* When we implement new providers with a corresponding test, the _sigar_ code will fail to compile on the target platform. This is an improvement because we generally don't know a provider hasn't been implemented (stub or otherwise) until we're in code that is a consumer of the sigar library. This leads to more platform specific handling when ideally sigar would take care of it all (or at least much more than it does).
* We can assert that the stub on an unsupported platform returns the correct `ErrNotImplemented` error, and ideally the test will fail when we do implement the new provider for that platform. I've tried to write the tests such that `ErrNotImplemented` is asserted against directly rather than just `HaveOccurred()`.
* This refactoring gives me an excuse to remove the `panic()` calls, because those are just difficult to deal with.

Note there is also some `gofmt` noise and other minor cleanup.

After this change, we should be able to hook up CI to this repository for all three platforms and know immediately when a PR is missing unsupported platform stubs or the like.

### CODE REVIEW

File: sigar_windows_test.go [hunk 1]
```diff
@@ -101,25 +99,21 @@ var _ = Describe("SigarWindows", func() {
...
 		It("gets TCP IPv6 conns", func() {
 			connList := sigar.NetTcpV6ConnList{}
 			err := connList.Get()
 			Ω(err).ShouldNot(HaveOccurred())
-			fmt.Printf("Connections: %v\n", connList)
 		})
 		It("gets UDP IPv6 conns", func() {
 			connList := sigar.NetUdpV6ConnList{}
 			err := connList.Get()
 			Ω(err).ShouldNot(HaveOccurred())
```
Comment (matthew-dailey) [inline sigar_windows_test.go:116]: Does `gomega` really use the capital omega for it's matchers?  I guess that must be how everyone else in the world feels about code using the latin alphabet :P

File: sigar_interface.go [hunk 2]
```diff
@@ -15,6 +16,8 @@ type Sigar interface {
 	GetSystemDistribution() (SystemDistribution, error)
 }
 
+var ErrNotImplemented error = errors.New("Collection not implemented")
```
Comment (matthew-dailey) [inline sigar_interface.go]: Do you want to specifically say "Collection not implemented _for this operating system_"?

File: concrete_sigar_test.go [hunk 3]
```diff
@@ -32,7 +35,9 @@ var _ = Describe("ConcreteSigar", func() {
 			firstValue := <-samplesCh
 
 			secondValue := <-samplesCh
-			Expect(secondValue.User).To(BeNumerically("<", firstValue.User))
+			if runtime.GOOS != "darwin" {
+				Expect(secondValue.User).To(BeNumerically("<", firstValue.User))
+			}
```
Comment (matthew-dailey) [inline concrete_sigar_test.go]: Can you put any checks in the `else` blocks for when it is `darwin`?

**Review by matthew-dailey — APPROVED** [review]
This refactor looks good @adamlamar, +1!  Are you worried that, having run `gofmt`, it may be harder to bring in upstream changes?  I guess any changes you'd want to bring in could also be `gofmt`'d to make the diff cleaner.

File: sigar_windows_test.go (hunk 1 above)
Comment (adamlamar) [reply sigar_windows_test.go:116]: Yeah, the omega character is <a href="https://onsi.github.io/gomega/#making-assertions">one way to make assertions</a>. Its... interesting. The `Expect()` syntax is equivalent.

File: concrete_sigar_test.go (hunk 3 above)
Comment (adamlamar) [reply concrete_sigar_test.go]: Good idea - this way the test will fail when we implement the CPU collection on darwin.

**Comment by adamlamar** [comment]
Thanks for the review @matthew-dailey!

We forked the repo in 2015 but we're only 4 commits behind, so I'm not too worried about merging upstream changes. Plus we've made a number of other changes that have caused it to diverge.

**Pull request merged**